packages/frontend-core/src/components/grid/cells/SignatureCell.svelte

```diff
@@ -0,0 +1,172 @@
+<script>
+  import { onMount, getContext } from "svelte"
+  import { SignatureModal } from "@budibase/frontend-core/src/components"
+  import { CoreSignature, ActionButton } from "@budibase/bbui"
+
+  export let schema
+  export let value
+  export let focused = false
+  export let onChange
+  export let readonly = false
+  export let api
+  export let invertX = false
+  export let invertY = false
+
+  const { API, notifications, props } = getContext("grid")
+
+  let isOpen = false
+  let modal
+
+  $: editable = focused && !readonly
+  $: {
+    if (!focused) {
+      close()
+    }
+  }
+
+  const onKeyDown = () => {
+    return false
+  }
+
+  const open = () => {
+    isOpen = true
+  }
+
+  const close = () => {
+    isOpen = false
+  }
+
+  const deleteSignature = async () => {
+    onChange(null)
+  }
+
+  const saveSignature = async sigCanvas => {
+    const signatureFile = sigCanvas.toFile()
+
+    let attachRequest = new FormData()
+    attachRequest.append("file", signatureFile)
+
+    try {
+      const uploadReq = await API.uploadBuilderAttachment(attachRequest)
+      const [signatureAttachment] = uploadReq
+      onChange(signatureAttachment)
+    } catch (error) {
+      $notifications.error(error.message || "Failed to save signature")
+      return []
+    }
+  }
+
+  onMount(() => {
+    api = {
+      focus: () => open(),
+      blur: () => close(),
+      isActive: () => isOpen,
+      onKeyDown,
+    }
+  })
+</script>
+
+<!-- svelte-ignore a11y-no-static-element-interactions -->
+<!-- svelte-ignore a11y-click-events-have-key-events -->
+<div
+  class="signature-cell"
+  class:light={!$props?.darkMode}
+  class:editable
+  on:click={editable ? open : null}
+>
+  {#if value?.url}
+    <!-- svelte-ignore a11y-missing-attribute -->
+    <img src={value?.url} />
+  {/if}
+</div>
+
+<SignatureModal
+  onConfirm={saveSignature}
+  title={schema?.name}
+  {value}
+  darkMode={$props.darkMode}
+  bind:this={modal}
+/>
+
+{#if isOpen}
```

Comment:
I think this has probably changed since you originally wrote this, but we now use a `GridPopover` component for any popovers like this. It was an important change to allow overflow going outside the grid.

Updating this is pretty straightforward thankfully - take a look at the attachment cell for an example. Basically just copy that exact usage of the GridPopover and put this stuff inside it (adjusting styles as needed).